rwatch/ui/animation/property_animation.c

```diff
@@ -0,0 +1,172 @@
+/* property_animation.c
+ * a convenience animation for 
+ * libRebbleOS
+ *
+ * Author: Carson Katri <me@carsonkatri.com>
+ */
+
+#include "librebble.h"
+#include "appmanager.h"
+#include "FreeRTOS.h"
+#include "property_animation.h"
+#include "animation.h"
+
+void property_animation_update_grect(PropertyAnimation * property_animation, const uint32_t distance_normalized)
+{
+    int progress = (distance_normalized * 100) / ANIMATION_NORMALIZED_MAX;
+    
+    Layer *layer = (Layer *) property_animation->subject;
+    
+    GRect *from = &property_animation->values.from.grect;
+    GRect *to = &property_animation->values.to.grect;
+    
+    GRect new_rect = GRect(from->origin.x + (((to->origin.x - from->origin.x) / 100.0)*progress), from->origin.y + (((to->origin.y - from->origin.y) / 100.0)*progress), from->size.w + (((to->size.w - from->size.w) / 100.0)*progress), from->size.h + (((to->size.h - from->size.h) / 100.0)*progress));
```

Comment:
Better to multiply through by progress first, then divide by 100.  Floating point arithmetic is very expensive.  i.e., `(to->origin.x - from->origin.x) * progress / 100`.  Or, better, to just integrate the `progress` change there: `(to->origin.x - from->origin.x) * distance_normalized / ANIMATION_NORMALIZED_MAX` (which works, because we know that `x * ANIMATION_NORMALIZED_MAX` will not be larger than a `uint32_t`.)

One way that I did made this easier in Dali Pebble was to define a linear-interpolation macro, along the lines of:

```
#define LERP(a, b)  ((a) + ((b) - (a)) * distance_normalized / ANIMATION_NORMALIZED_MAX)
```

Which can then be used later as: `GRect(LERP(from->origin.x, to->origin.x), LERP(from->origin.y, to->origin.y) ...)`.